### PR TITLE
Migrate old style API connectors

### LINKED ISSUE #5250: Migrate old style API connectors
## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
When importing an integration with a API connector, the connector and and the OpenAPI specification get imported into Syndesis, but building the integration using this connector fails when assembling.

Also if this is meant to happen, something should indicate inside Syndesis that the integration is in invalid state because the integration status also changes from **starting** to **building** even though there is no build pod present. Which makes the user just what is taking so long.

Sample integration: [https://drive.google.com/open?id=1l9A5lS8l1EdsyyVSaj7bej0BFlOw1Rk-](https://drive.google.com/open?id=1l9A5lS8l1EdsyyVSaj7bej0BFlOw1Rk-)

Stacktrace:
```
java.lang.UnsupportedOperationException: Old style of connectors from camel-connector are not supported anymore, please be sure that integration json satisfy connector.getComponentScheme().isPresent() || descriptor.getComponentScheme().isPresent()
	at io.syndesis.integration.project.generator.ProjectGenerator.generateApplicationProperties(ProjectGenerator.java:174) ~[integration-project-generator-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at io.syndesis.server.controller.integration.online.PublishHandler.createDeploymentData(PublishHandler.java:136) ~[server-controller-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at io.syndesis.server.controller.integration.online.PublishHandler.execute(PublishHandler.java:99) ~[server-controller-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at io.syndesis.server.controller.StateChangeHandler.execute(StateChangeHandler.java:33) [server-controller-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at io.syndesis.server.controller.integration.BaseIntegrationController.lambda$callStateChangeHandler$10(BaseIntegrationController.java:220) [server-controller-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_151]
```

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #5250

### CODE REVIEW

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+migrate("connectors", "/connectors", migrateConnector);
+migrate("connections", "/connections", function(connection) {
+    return migrateConnector(connection.connector);
+});
+migrate("integrations", "/integrations", function(integration) {
+    var done = false
+
+    if (integration.steps) {
+        integration.steps.forEach(function (step) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
        integration.steps.forEach(step => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            console.log("")
+            done |= migrateStep(step)
+        });
+    }
+
+    if (integration.flows) {
+        integration.flows.forEach(function (flow) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
        integration.flows.forEach(flow => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            flow.steps.forEach(function (step) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
            flow.steps.forEach(step => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+//
+
+console.log("Migration to schema 29 ...");
+console.log("This migration will update API Client to new connector model");
+
+migrate("connectors", "/connectors", migrateConnector);
+migrate("connections", "/connections", function(connection) {
+    return migrateConnector(connection.connector);
+});
+migrate("integrations", "/integrations", function(integration) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
migrate("integrations", "/integrations", integration => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+//
+// Migration
+//
+
+console.log("Migration to schema 29 ...");
+console.log("This migration will update API Client to new connector model");
+
+migrate("connectors", "/connectors", migrateConnector);
+migrate("connections", "/connections", function(connection) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
migrate("connections", "/connections", connection => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+var migrate = function(type, path, consumer) {
+    console.log("Start " + type + " migration")
+
+    var migrated  = 0;
+    var inspected = 0;
+    var elements  = jsondb.get(path);
+
+    if (elements) {
+        Object.keys(elements).forEach(function(elementId) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
        Object.keys(elements).forEach(elementId => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        ];
+        connector.dependencies = [
+            {
+                "id": "io.syndesis.connector:connector-rest-swagger",
+                "type": "MAVEN"
+            }
+        ];
+
+        if (connector.actions) {
+            connector.actions.forEach(function (action) {
```
Comment (christophd) [inline app/server/runtime/src/main/resources/migrations/up-29.js]:
```suggestion
            connector.actions.forEach(action => {
```

File: app/server/runtime/src/main/resources/migrations/up-29.js (hunk 1 above)
Comment (lburgazzoli) [reply app/server/runtime/src/main/resources/migrations/up-29.js]: nashorn does not support => afaik

Comment (christophd) [reply app/server/runtime/src/main/resources/migrations/up-29.js]: I did not know that. You seem to be right. At least for Java 8 I did not find any examples for that support. If that's so of course my comments are obsolete.

Comment (lburgazzoli) [reply app/server/runtime/src/main/resources/migrations/up-29.js]: afaik, this will be supported by graaljs

**Pull request merged**